Documentation/TileMap.md

```diff
@@ -0,0 +1,39 @@
+# Why do we use TileMap.h?
```

Comment:
Why ```TileMap.h```? Can I use this library with C or it's only C++?
We need a pattern. I recommend hpp, because we are using C++. :warning:
Nice documentation! :+1: